### PR TITLE
remove a redundant check on PGA_WRITEABLE

### PR DESCRIPTION
**Pull request by wy-chung**
"#!pmap_page_is_write_mapped(m)#" is equal to "(m->a.flags & PGA_WRITEABLE) == 0" so one of the if statements can be removed.

### CODE REVIEW

File: sys/riscv/riscv/pmap.c [hunk 1]
```diff
@@ -4611,19 +4611,17 @@ pmap_clear_modify(vm_page_t m)
...
 	vm_page_assert_busied(m);
 
-	if (!pmap_page_is_write_mapped(m))
-	        return;
-
 	/*
 	 * If the page is not PGA_WRITEABLE, then no PTEs can have PG_M set.
 	 * If the object containing the page is locked and the page is not
 	 * exclusive busied, then PGA_WRITEABLE cannot be concurrently set.
 	 */
```
Comment (mhorne) [inline sys/riscv/riscv/pmap.c:4621]:
```suggestion

```
This comment should be removed, it is no longer accurate and not present in other pmap implementations.

In fact it seems this comment and check were missed in the previous change, see: 638f867814.

**Review by mhorne — CHANGES_REQUESTED** [review]
Thanks for finding this! A tweak is required but the change is good.

Please confirm or adjust the Author field of the commit.
Currently it reads: `wy-chung <wy-chung@outlook.com>`

File: sys/riscv/riscv/pmap.c (hunk 1 above)
Comment (bsdimp) [reply sys/riscv/riscv/pmap.c:4621]: Should we do this here, or as a separate commit?

Comment (wy-chung) [reply sys/riscv/riscv/pmap.c:4621]: The user.name and user.email are correct.

**Review by mhorne — APPROVED** [review]

**Comment by mhorne** [comment]
Thanks. Merged to `main` as 2eb64df7f53d6. This PR can be closed.

**Pull request closed without merging**